conda-store-server/conda_store_server/server/auth.py

```diff
@@ -233,12 +237,24 @@ def is_subset_entity_permissions(self, entity, new_entity):
                 return False
         return True
 
-    def authorize(self, entity, arn, required_permissions):
+    def authorize(
+        self, entity: schema.AuthenticationToken, arn: str, required_permissions
+    ):
         return required_permissions <= self.get_entity_permissions(
             entity=entity, arn=arn
         )
 
-    def database_role_bindings(self, entity):
+    def database_role_bindings(self, entity: schema.AuthenticationToken):
+        # This method can be reached from the router_ui via filter_environments.
+        # Since the UI routes are not versioned, we don't know which API version
+        # the client might be using. So we check all API versions, starting from
+        # the most recent one.
+        v2 = self._database_role_bindings_v2(entity)
+        if v2:
+            return v2
+        return self._database_role_bindings_v1(entity)
```

Comment:
I don't like this solution anymore. Once there's any data in v2 (even if added by mistake), v1 will never be checked. I'll add a CLI parameter to check a particular table here. The other one will be ignored.

Reply:
Resolved by adding an overridable field to the RBAC class: `RBACAuthorizationBackend.role_mappings_version`.